x-pack/plugins/observability_solution/dataset_quality/public/components/dataset_quality_details/degraded_field_flyout/possible_mitigations/manual/index.tsx

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { EuiLink, EuiPanel, EuiSpacer, EuiText, EuiCopy, EuiButtonEmpty } from '@elastic/eui';
+import { useKibanaContextForPlugin } from '../../../../../utils';
+import {
+  manualMitigationCustomPipelineText,
+  otherMitigationsCustomComponentTemplate,
+  otherMitigationsCustomIngestPipeline,
+} from '../../../../../../common/translations';
+import { useDatasetQualityDetailsState } from '../../../../../hooks';
+import { getComponentTemplatePrefixFromIndexTemplate } from '../../../../../../common/utils/component_template_name';
+
+export function ManualMitigations() {
+  const { integrationDetails } = useDatasetQualityDetailsState();
+  const isIntegration = !!integrationDetails?.integration;
+  return (
+    <>
+      <EditComponentTemplate isIntegration={isIntegration} />
+      <EuiSpacer size="s" />
+      <EditPipeline isIntegration={isIntegration} />
+    </>
+  );
+}
+
+function EditComponentTemplate({ isIntegration }: { isIntegration: boolean }) {
+  const {
+    services: {
+      share: {
+        url: { locators },
+      },
+    },
+  } = useKibanaContextForPlugin();
+
+  const { dataStreamSettings, datasetDetails } = useDatasetQualityDetailsState();
+  const { name } = datasetDetails;
+
+  const componentTemplateUrl = locators.get('MANAGEMENT_APP_LOCATOR')?.useUrl(
+    isIntegration
+      ? {
+          componentTemplate: `${getComponentTemplatePrefixFromIndexTemplate(
+            dataStreamSettings?.indexTemplate ?? name
+          )}@custom`,
+        }
+      : { indexTemplate: dataStreamSettings?.indexTemplate }
+  );
```

Comment:
> [!WARNING]
> This won't work as expected because useUrl is missing the dependencies list that would be passed to the internal `useEffect`, in this case it should be `[isIntegration]`.
> However, since it makes more contrived and less readable, I would suggest a different approach that doesn't required side effects changes on the property change:
```ts
const customComponentTemplateUrl = locators.get('MANAGEMENT_APP_LOCATOR')?.useUrl({
  componentTemplate: `${getComponentTemplatePrefixFromIndexTemplate(
    dataStreamSettings?.indexTemplate ?? name
  )}@custom`,
});

const indexTemplateUrl = locators
  .get('MANAGEMENT_APP_LOCATOR')
  ?.useUrl({ indexTemplate: dataStreamSettings?.indexTemplate });

const componentTemplateUrl = isIntegration ? customComponentTemplateUrl : indexTemplateUrl;
```
This marks a clear separation and the each URL is correctly computed once.